turbinia/processors/archive.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Processor for compressing and decompressing directories."""
+
+from __future__ import unicode_literals
+
+import os
+import tarfile
+import logging
+
+from turbinia import TurbiniaException
+
+log = logging.getLogger('turbinia')
+
+
+def CompressFolder(local_path):
+  """Compress a given directory into a gzip file.
+
+  Args:
+    local_path(str): The path to the directory.
+
+  Returns:
+    str: The path to the gzip file.
+  """
+  # Check if the File or Directory exists.
+  if not os.path.exists(local_path):
+    raise TurbiniaException(
+        'The File or Directory does not exist: {0:s}'.format(local_path))
+
+  # Iterate through a given list of files and compress them.
+  file_names = os.listdir(local_path)
```

Comment:
Ah yeah you are right. I did test compression against a folder containing nested directories and files contained within these nested directories and it would still compress these files. I could also just not do a listdir() and compress the root directory instead, but it compresses it with the absolute path. For example, if the directory was in /usr/path/to/files/file_to_compress, when you compress the directory "file_to"compress", it will also embed the previous paths, "usr", "path", "to", "files" so you'd need to navigate through all of them to get to the directory you actually compressed. With how I implemented it, running listdir() then adding the base folders will keep that directory structure.